From: Simon McVittie <smcv@debian.org>
Date: Wed, 26 Jun 2019 09:54:59 +0100
Subject: tests: Skip tests that use system helper if uid or gid is 0

The system helper refuses to operate in test mode while privileged
(because that would result in it skipping authentication, which would
be very bad), but some CI systems build in a container or VM as uid 0
or gid 0.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/libtest.sh     | 7 +++++++
 tests/test-config.sh | 2 ++
 tests/test-info.sh   | 4 ++++
 3 files changed, 13 insertions(+)

diff --git a/tests/libtest.sh b/tests/libtest.sh
index 4a5e699..365eb2b 100644
--- a/tests/libtest.sh
+++ b/tests/libtest.sh
@@ -334,6 +334,13 @@ else
     _flatpak_bwrap_works=true
 fi
 
+skip_if_root () {
+    if [ "x$(id -u)" = x0 ] || [ "x$(id -g)" = x0 ]; then
+        echo "1..0 # SKIP system helper refuses to run in test mode as root"
+        exit 0
+    fi
+}
+
 skip_without_bwrap () {
     if "${_flatpak_bwrap_works}"; then
         return 0
diff --git a/tests/test-config.sh b/tests/test-config.sh
index 420f898..e275d1f 100755
--- a/tests/test-config.sh
+++ b/tests/test-config.sh
@@ -24,6 +24,8 @@ set -euo pipefail
 # This test looks for specific localized strings.
 export LC_ALL=C
 
+skip_if_root
+
 echo "1..5"
 
 ${FLATPAK} config --list > list_out
diff --git a/tests/test-info.sh b/tests/test-info.sh
index cb5fdb7..7885d2f 100644
--- a/tests/test-info.sh
+++ b/tests/test-info.sh
@@ -4,6 +4,10 @@ set -euo pipefail
 
 . $(dirname $0)/libtest.sh
 
+if [ x${USE_SYSTEMDIR-} == xyes ] ; then
+    skip_if_root
+fi
+
 echo "1..7"
 
 setup_repo
